pkg/autoscaler/statforwarder/forwarder.go

```diff
@@ -48,11 +48,22 @@ const (
 	autoscalerPortName = "http"
 	retryTimeout       = 3 * time.Second
 	retryInterval      = 100 * time.Millisecond
+
+	// Retry as most 15 seconds to process a stat. NOTE: Retrying could
+	// cause high delay and inaccurate scaling decision so we use
+	// the timestamp on receiving.
+	maxProcessingRetry      = 30
+	retryProcessingInterval = 500 * time.Millisecond
```

Comment:
I sort of feel like we should either do a PollImmediate (if this is going to be a very short-lived retry), or we should just use a workqueue.RateLimitingInterface, since we're kind of ending up building one from channels here anyway.

Reply:
Please see https://github.com/knative/serving/pull/9404#issuecomment-692918059

Reply:
given we're using a reenqueue approach (and I get the reasoning for that), just to check: any particular reason we're doing it with channels rather than using queue.DelayingInterface/queue.RateLimitingInterface from client-go?

Reply:
I was not aware of queue.DelayingInterface/queue.RateLimitingInterface. Any benefit to use them?

Reply:
just that I think they might do a lot of this stuff for us, and they let us do configurable back-offs and whatnot. Also we already use them in the stat prober and the new digest resolver

Reply:
Our use case here is pretty simple: delay and retrying. I feel it's more complicated to use queue.DelayingInterface/queue.RateLimitingInterface as you have to explicitly call `Done` and `Forgot` functions.

Reply:
ha, funny you should say that -- just realised we forgot to do that in the image digest resolver!

Reply:
let's go with this, and we can always adopt the queue stuff if we need it later -- agree with your thinking it's probably overkill for now

Reply:
:)